common/src/main/scala/org/apache/celeborn/common/meta/WorkerInfo.scala

```diff
@@ -36,6 +36,10 @@ class WorkerInfo(
     val pushPort: Int,
     val fetchPort: Int,
     val replicatePort: Int,
+    val internalPort: Int,
```

Comment:
I don't think we need to add these secure ports(It looks a little complicated beside insecure ports), IMO the celeborn service is either open securely or not. It seems strange to keep secure connection beside with insecure connection, if so it's still insecure for celeborn service?

For users who want to use security features, they can create a new Celeborn service to instead and finally migrate other tasks to it.

Reply:
@RexXiong Interesting, so you mean that we don't need to support seamless rolling upgrade from previously unsecured cluster to secured cluster, and one cluster can be either secured or unsecured, not both. Supporting mixed mode for rolling upgrade does introduces complexity. WDYT @otterc @mridulm @pan3793 ?

Reply:
In our last sync about authentication on September 23rd, we concluded, based on community feedback, that we would add separate ports for secured communication to allow for seamless rolling upgrades. In that meeting, we decided that it would be akin to supporting both HTTP and HTTPS by a web server. Here is the link to the summary: https://www.mail-archive.com/dev@celeborn.apache.org/msg00618.html.

However, IMO, allowing both secured and unsecured communication adds complexity and creates the possibility of issues. There is a way to achieve seamless rolling upgrades without adding new secured ports to Celeborn: by deploying another cluster of Celeborn that is secured (with different configured ports for Master/Worker) and leaving the existing unsecured cluster as it is. However, this creates more toil for the deployment/monitoring infrastructure.

We are not currently running Celeborn in production, and we will only run it in production with authentication enabled. Therefore, seamless rolling upgrades are not a problem for us, but we were addressing the community's concern.

If the community agrees that adding dedicated secured ports is not something we want to proceed with, I am willing to make those changes.

cc. @waitinfuture @mridulm @pan3793 @RexXiong

Reply:
Thanks @otterc for the explanation. I agree with you that 

> allowing both secured and unsecured communication adds complexity and creates the possibility of issues

, and I believe @RexXiong have the same opinion. IMO whether to support rolling upgrade should depend on whether auth is enabled, as the following form describes:

| Auth enabled before  | Auth enabled after | Should support rolling upgrade |
| ------------- | ------------- | ------------- |
| No  | No  | Yes |
| No  | Yes  | No |
| Yes  | Yes  | Yes |
| Yes  | No  | No |

In summary, we should support rolling upgrade only if auth configuration stays the same before and after, and don't support otherwise.

WDYT? @otterc @RexXiong @mridulm @pan3793

Reply:
Thanks @waitinfuture @otterc, sorry I may have lost track of some aspects of the original design, but this really adds a lot of complexity, I agree with summary about `we should support rolling upgrade only if auth configuration stays the same`

Reply:
it seems there is more complexity than I first thought.

given the above discussion plus feedback from Celeborn users in other threads, most community members tend to downgrade the backward compatibility to "support rolling upgrade only if auth configuration stays the same", so I'm fine with this decision

Reply:
Thanks, as @otterc mentioned, this will significantly reduce the complexity !

Reply:
Sounds good to me. Will remove the additional secure ports. We have already committed a secure RPC port for Master for which I will create a separate PR to remove. Please note that we do need internal ports for the servers to communicate with each other. The internal ports allow internal communication to follow a different flow than the external. However, the internal port number doesn't need to be part of the WorkerInfos equality.

Reply:
I have made the necessary changes. Please take a look.